From the clang-smoketest [failure](https://github.com/kokkos/kokkos/actions/runs/24425631490/job/71358845139?pr=9075):
```
 /home/runner/work/kokkos/kokkos/kokkos/core/src/Kokkos_View.hpp:1491:9: error: unused type alias 'rhs_view' [-Werror,-Wunused-local-typedef]
 1491 |   using rhs_view   = View<RT, RP...>;
```

--- a/core/src/Kokkos_View.hpp
+++ b/core/src/Kokkos_View.hpp
@@ -1496,7 +1496,7 @@ KOKKOS_INLINE_FUNCTION bool operator==(const View<LT, LP...>& lhs,
                         typename rhs_traits::array_layout> &&
          std::is_same_v<typename lhs_traits::memory_space,
                         typename rhs_traits::memory_space> &&
-         View<LT, LP...>::rank() == View<RT, RP...>::rank() &&
+         lhs_view::rank() == rhs_view::rank() &&
          lhs.data() == rhs.data() && lhs.span() == rhs.span() &&
          Kokkos::Impl::view_equal_extents_impl(
              lhs, rhs, std::make_index_sequence<lhs_view::rank()>{});

--- a/core/src/Kokkos_View.hpp
+++ b/core/src/Kokkos_View.hpp
@@ -1496,8 +1496,8 @@ KOKKOS_INLINE_FUNCTION bool operator==(const View<LT, LP...>& lhs,
                         typename rhs_traits::array_layout> &&
          std::is_same_v<typename lhs_traits::memory_space,
                         typename rhs_traits::memory_space> &&
-         lhs_view::rank() == rhs_view::rank() &&
-         lhs.data() == rhs.data() && lhs.span() == rhs.span() &&
+         lhs_view::rank() == rhs_view::rank() && lhs.data() == rhs.data() &&
+         lhs.span() == rhs.span() &&
          Kokkos::Impl::view_equal_extents_impl(
              lhs, rhs, std::make_index_sequence<lhs_view::rank()>{});
 }

--- a/core/unit_test/view/TestViewEqualityOperator.hpp
+++ b/core/unit_test/view/TestViewEqualityOperator.hpp
@@ -113,14 +113,12 @@ void test_view_equality_operator() {
   }
 
   {
-    using v_t =
-        Kokkos::View<T[3][5], Kokkos::LayoutLeft, TEST_EXECSPACE>;
-    using col_t =
-        Kokkos::View<T[3], Kokkos::LayoutLeft, TEST_EXECSPACE,
-                     Kokkos::MemoryUnmanaged>;
+    using v_t   = Kokkos::View<T[3][5], Kokkos::LayoutLeft, TEST_EXECSPACE>;
+    using col_t = Kokkos::View<T[3], Kokkos::LayoutLeft, TEST_EXECSPACE,
+                               Kokkos::MemoryUnmanaged>;
 
     v_t v_0("A_layoutleft_extent_rank_eq");
-    auto col_0    = Kokkos::subview(v_0, Kokkos::ALL, 0);
+    auto col_0 = Kokkos::subview(v_0, Kokkos::ALL, 0);
     col_t alias_to_col_0(col_0.data());
     ASSERT_TRUE(check_equal(col_0, alias_to_col_0));
   }

--- a/core/unit_test/view/TestViewEqualityOperator.hpp
+++ b/core/unit_test/view/TestViewEqualityOperator.hpp
@@ -111,6 +111,19 @@ void test_view_equality_operator() {
         v_0.data());
     ASSERT_TRUE(check_equal(v_0, v_1));
   }
+
+  {
+    using v_t =
+        Kokkos::View<T[3][5], Kokkos::LayoutLeft, TEST_EXECSPACE>;
+    using col_t =
+        Kokkos::View<T[3], Kokkos::LayoutLeft, TEST_EXECSPACE,
+                     Kokkos::MemoryUnmanaged>;
+
+    v_t v_0("A_layoutleft_extent_rank_eq");
+    auto col_0    = Kokkos::subview(v_0, Kokkos::ALL, 0);
+    col_t alias_to_col_0(col_0.data());
+    ASSERT_TRUE(check_equal(col_0, alias_to_col_0));
+  }
 }
 
 TEST(TEST_CATEGORY, view_equality_operator) { test_view_equality_operator(); }

